### ISSUE TITLE
[Q] Not available videos become available as soon as I proxy through the same machine

### ISSUE THREAD

**Issue by delivrance** [rg3/youtube-dl#18456]
### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2018.12.03*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2018.12.03**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [x] Question
- [ ] Other

### Description of your *issue*, suggested solution and other information

I'm getting `This video is not available` errors for some videos lately using, for example:
`youtube-dl -F <video_id>`.

The weird thing is that the same video_ids work just fine if I use a proxy which is actually running on the *same machine*:
`youtube-dl --proxy http://user:pass@localhost:1234 -F <video_id>`

I tried with `--rm-cache-dir`, installed again, rebooted... Any other ideas?

**Comment by dstftw** [comment]
Anything from geo restriction to mis-copy-pasted video id.

**Comment by dstftw** [comment]
Verbose logs with `--dump-pages --print-traffic`.

**Comment by delivrance** [comment]
@dstftw [No proxy](https://hastebin.com/yufitigedi.txt) (non-working). [With proxy](https://hastebin.com/arofidawuv.txt) (working)

**Comment by dstftw** [comment]
Does http://yt-dl.org/ip shows the same IPs?

**Comment by delivrance** [comment]
I am indeed getting different IPs. It turns out that connections through the proxy are done via IPv6 instead; `youtube-dl -F -6 <video_id>` workarounds my issue, but still I have no idea why *only some* video fail on IPv4. Probably youtube-side issues?

**Comment by dstftw** [comment]
Well, looking at first webpages from both runs it clearly seen that both cases are served with *video-not-available* webpage (video can't be watched in browser in both cases). But in case of IPv6 `ytplayer.config` is still included in webpage's source making it available to download by youtube-dl. So it looks like a Youtube's bug and a loophole for youtube-dl at the same time.

**Comment by delivrance** [comment]
Alright, thanks. Let's see if something changes in the next days, I'll eventually report back here.

What made me say I'm able to watch them in browser was an actual test I did by proxying my local machine through that server, and videos were playable.

One more little question while we are at it: my application is actually embedding youtube-dl in a python script, so I would need to force IPv6 in there, but I couldn't find anything related to IPv6 nor IPv4 [here](https://github.com/rg3/youtube-dl/blob/3e4cedf9e8cd3157df2457df7274d0c842421945/youtube_dl/YoutubeDL.py#L137-L312). I tried `forceipv6`, `force-ipv6`, `6`, `ipv6` and some other variations `= True`.
